### PR TITLE
feat: repositories page styling updates with extended BranchSelector functionality

### PR DESCRIPTION
**Pull request by andrewgolovanov**
**Description:**

This pull request introduces the following changes:

- **Enhancement and revision of the color palette**: Expands and adjusts the existing color palette to align with updated design requirements.
- **Refactoring, expansion, and bug fixes in component styles**: Updates styles for multiple components, including `Badge`, `Button`, `BranchesList`, `Pagination`, `SearchBox`, `Skeleton`, `StakedList`, `DropdownMenu`, `Tabs`, and `Text` to improve consistency and usability.
- **Styling updates and fixes for the Repositories page**: Implements design-based updates and corrections to enhance visual consistency. – [Preview](https://canary-git-pixelpoint-dev-pixelpoint.vercel.app/sandbox/repos)
- **Extension of functionality for the Branch Selector component**: Adds new features and improves the component's functionality. – [Preview](https://canary-git-pixelpoint-dev-pixelpoint.vercel.app/sandbox/repos/drone/summary)
- **Updating the main navbar** - refactoring and updating styles
- **New filtering system**:  that provides advanced filtering and sorting capabilities for repository lists. The implementation includes support for multiple filter types (`checkbox`, `calendar`, `text`, and `number`) with various conditions and sorting options. - [Preview](https://deploy-preview-416--harness-xd-review.netlify.app/sandbox/repos)

**Key Features of the Filter**
**Filter Types**
- **[Checkbox Filter](https://github.com/harness/canary/pull/416/files#diff-65805f5c344f67efc00678b9f56a5e30d00af9b76d1bb9e5a6d5575e6947c15eR1)**: Multi-select options with basic conditions (`is`, `is not`, `is empty`, `is not empty`)
- **[Calendar Filter](https://github.com/harness/canary/pull/416/files#diff-2378afb3e41b9ba14c97ab0fd10c458579a7dc8c7e324eea82c60c9bc35c8956R1)**: Date-based filtering with range support (`is`, `is before`, `is after`, `is between`)
- **[Text Filter](https://github.com/harness/canary/pull/416/files#diff-0011a61a7732f446b5a688e677bb28a75f31c9ca99e530c12cb4084496a58146R1)**: String operations (`contains`, `starts with`, `ends with`, etc.)
- **[Number Filter](https://github.com/harness/canary/pull/416/files#diff-b81c5e0ec2fe575a76b04f9629ed597a2445d70dd10528091068fa7a74da69b8R1)**: Numeric comparisons (`=`, `≠`, `>`, `<`, `≥`, `≤`)

**Core Components**
1. **[FiltersBar](https://github.com/harness/canary/pull/416/files#diff-aad7ffce24b652a3d9aaf50212c65f4f1f64fc592e4990aeddb5d16c5956dea1R1)**: Main container component that displays active filters and sorts
2. **[Filters](https://github.com/harness/canary/pull/416/files#diff-7d119027a2d687f9f4d18de6b2b60a014e465a1240ef65d789b2570e34f46a69R1)**: Handles filter selection and management
3. **[FilterTrigger](https://github.com/harness/canary/pull/416/files#diff-a4bf0cf890c6f7d0ae27c3359be670c273d27f7b275a37b3e0608bcd5c8e078eR1)**: UI component for filter/sort selection
4. **Filter Variants**: Specialized components for each filter type (checkbox, calendar, text, number)

**Functionality**
- Multiple active filters with AND logic
- Real-time filtering
- Search within filter options
- Clear individual filters or reset all
- Customizable filter conditions per type
- Sortable results with multiple criteria
- Drag-and-drop sort priority

**Screenshots:**
<details>
<summary>Repositories page</summary>
image
</details>

<details>
<summary>Branch Selector</summary>
image
</details>

<details>
<summary>Filter Trigger</summary>
Filter
image

Sort
image
</details>

<details>
<summary>Checkbox Filter</summary>
image
</details>

<details>
<summary>Calendar Filter</summary>
image
</details>

<details>
<summary>Text Filter</summary>
image
</details>

<details>
<summary>Number Filter</summary>
image
</details>

<details>
<summary>Sorts</summary>
image
</details>

<details>
<summary>Navbar</summary>
Arc 2024-11-18 13 19 30
Arc 2024-11-18 13 19 59
</details>

### CODE REVIEW

File: packages/canary/src/components/text.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ const textVariants = cva('text-base', {
       both: ''
     },
     truncate: {
-      true: 'truncate text-ellipsis text-nowrap'
```
Comment (andrewgolovanov) [inline packages/canary/src/components/text.tsx:36]: this class already contains the required properties, so they aren't necessary

Google Chrome 2024-11-11 17 05 32

File: packages/canary/src/components/text.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ const textVariants = cva('text-base', {
...
+      true: 'truncate'
```
Comment (andrewgolovanov) [inline packages/canary/src/components/text.tsx:36]: This class already contains the necessary properties such as `text-ellipsis text-nowrap`, so they aren't necessary

Google Chrome 2024-11-11 17 06 06

**Comment by abhinavrastogi-harness** [comment]
Currently this PR contains breaking changes in the contracts of some components. This will break Gitness. 
Let's discuss today around what process we will need to follow to manage these contract changes smoothly.

**Comment by bot2-harness** [comment]
This pull request has been automatically closed due to 30 days of inactivity. If you believe this was done in error, please feel free to reopen the PR or reach out to the BT team for further assistance. Thank you for your contributions!

**Pull request closed without merging**